src/lambda/commands/deploySamApplication.ts

```diff
@@ -324,3 +330,36 @@ function getDefaultWindowFunctions(): WindowFunctions {
         showInformationMessage: vscode.window.showInformationMessage,
     }
 }
+
+function notifyDuringLongDeployment(deployParameters: DeploySamApplicationParameters): NodeJS.Timeout {
+    return setTimeout(async () => {
+        const cfnClient = new DefaultCloudFormationClient(deployParameters.region)
+        const stack = (await cfnClient.describeStacks(deployParameters.destinationStackName)).Stacks
+        if (stack !== undefined) {
+            const helpLink = localize('AWS.samcli.deploy.stuckStackHelpLink', 'View docs')
+            const cfnConsole = localize('AWS.samcli.deploy.goToCfnDashboard', 'View Stack')
+            const viewAwsStatus = localize('AWS.samcli.deploy.viewAwsStatus', '{0} status', getIdeProperties().company)
+            const cancelUpdate = localize('AWS.samcli.deploy.cancelUpdate', 'Cancel update')
+            const cfnConsoleUrl = `https://${deployParameters.region}.console.aws.amazon.com/cloudformation/home?region=${deployParameters.region}`
+            const items = [helpLink, cfnConsole, viewAwsStatus]
+            if (stack[0].StackStatus === 'UPDATE_IN_PROGRESS') {
+                items.push(cancelUpdate)
+            }
+            const stackStuckMessage = localize(
+                'AWS.samcli.deploy.stackStuckWarning',
+                'Your deployment seems to be taking a while. If this is unexpected, it may be stuck in progress. You may continue to wait or choose an option below.'
```

Comment:
- can be less verbose (which is especially important in UI messages)
- "seems to be" is a phrase to avoid, it doesn't mean anything except "we have no idea"

```suggestion
                'Deployment is taking a while. You may continue to wait or choose an option below.'
```